extensions/functions_arithmetic.yaml

```diff
@@ -1563,6 +1563,43 @@ aggregate_functions:
             values: [ TIE_TO_EVEN, TIE_AWAY_FROM_ZERO, TRUNCATE, CEILING, FLOOR ]
         nullability: DECLARED_OUTPUT
         return: fp64?
+  - name: "first"
```

Comment:
the window functions are called `first/last_value`, should we stick to that naming? 

Spark seems to have both `first` and `first_value`, though first_value is only supported in SQL while first is supported as a method. 
DataFusion has `first_value`

Reply:
Looking at other engines, Postgres and Trino only have support for `first_value` and `last_value` as window functions, and don't have `first` and `last` aggregate functions.

I think it make sense to keep first/last and first_value/last_value as seperate.

Reply:
Yeah, looks like this is a mess overall - postgres has only `first_value` as window, Spark has `first` and `first_value` being the same, duckdb has `first` for aggregate and both for window, DataFusion has `first_value` as aggregate...

The purpose of the functions is the same, though. I actually think it'd make sense for Substrait to only have one set of these (as aggregate), and maybe it should be the `_value` option to match already existing `any_value`. But dunno if we can remove the window versions, I guess that'd be a breaking change?

Reply:
I renamed them to have the "_value" postfix now - small annoyance there is that if someone includes now both these and window functions they'll get duplicate signatures...

Reply:
Currently the base name has to be unique across all of the files.  So having two first_value functions with the same signature will likely mess things up.  (And yes, we need a check.)

Reply:
> Currently the base name has to be unique across all of the files

I just saw this comment. I believe this is inconsistent with both the spirit and spec of extensions. Extensions should allow people to create new extensions that completely conflict with other extensions. That's why the spec specifies that functions are identified by a combination of their name and URI. Two different URIs can define the same names as entirely different things since they are in different namespaces.

Reply:
Recent discussion on the topic:

https://github.com/substrait-io/substrait/pull/631#issuecomment-2065325055
https://github.com/substrait-io/substrait/issues/634

IIRC, the general consensus is "yes, different filenames should be able to have duplicate functio names but we're not there yet, there isn't much motivation to tackle the issue, and keeping the core Substrait functions unique makes life easier in the short term"